ZipEditor/src/zipeditor/model/ZipModel.java

```
@@ -102,10 +106,24 @@ public static ContentTypeId detectType(InputStream contents) {
 			contents.reset();
 			if (count == -1)
 				return null;
-			ZipInputStream zip = new ZipInputStream(contents);
-			if (zip.getNextEntry() != null) {
-				contents.reset();
-				return ContentTypeId.ZIP_FILE;
+			try {
+				ZipArchiveInputStream zip = new ZipArchiveInputStream(contents) {
+					
+					@Override
+					protected InputStream createZstdInputStream(InputStream in) throws IOException {
+						return ZstdUtilities.getInputStream(in);
+					}
+					
+				};
+				if (zip.getNextEntry() != null) {
+					contents.reset();
+					return ContentTypeId.ZIP_FILE;
+				}
+			} catch (IOException ioe) {
+				if (ioe instanceof ZipEditorZstdException) {
+					return ContentTypeId.INVALID;
+				}
+				// thrown in getNextEntry() method if, if file is not a zip file.
```

Not sure I understood this change & comment.
So if the exception is **not** `ZipEditorZstdException` we **silently continue**? Shouldn't we re-throw `IOException` here, or, better, catch `ZipEditorZstdException` only?

---

Yes.. It tries to identify the next archive type.. But if the ZipEditorZstdException is thrown, we know that we can't open the archive anymore (due to missing library or deactivated zstd support). See line 140 where the next archive check was missed and it is about to try the next archive type.

Added a comment and fixed the comment in line 126.